app/src/main/java/io/intrepid/contest/screens/sendinvitations/SendInvitationsPresenter.java

```diff
@@ -2,48 +2,116 @@
 
 import android.support.annotation.NonNull;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import io.intrepid.contest.R;
 import io.intrepid.contest.base.BasePresenter;
 import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Contact;
 
 public class SendInvitationsPresenter extends BasePresenter<SendInvitationsContract.View>
         implements SendInvitationsContract.Presenter {
 
+    private final List<Contact> fullContactList = new ArrayList<>();
+    private final List<Contact> selectedContactList = new ArrayList<>();
+    private boolean contactSelectionEnabled = false;
+    private boolean displayMenuItems = true;
+
     public SendInvitationsPresenter(@NonNull SendInvitationsContract.View view,
                                     @NonNull PresenterConfiguration configuration) {
         super(view, configuration);
     }
 
     @Override
+    public boolean isContactSelectionEnabled() {
+        return contactSelectionEnabled;
+    }
+
+    private void setContactSelectionEnabled(boolean enabled) {
+        this.contactSelectionEnabled = enabled;
+    }
+
+    @Override
+    public List<Contact> getContactList() {
+        if (isContactSelectionEnabled()) {
+            return fullContactList;
+        }
+        return selectedContactList;
+    }
+
+    @Override
+    public boolean hasContactPermissions() {
+        return view.checkContactsPermissions();
+    }
+
+    @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+
+        showPreviewContent();
+    }
+
+    @Override
     protected void onViewBound() {
         super.onViewBound();
 
-        if (!view.hasContactsPermissions()) {
+        if (!hasContactPermissions()) {
```

Comment:
Yeah, it felt like a lot of calls to the view, but as permission changes are out of presenter control...